opensearch-service-domain-cdk/lib/opensearch-service-domain-cdk-stack.ts

```diff
@@ -0,0 +1,38 @@
+import { Construct } from 'constructs';
+import { EbsDeviceVolumeType } from "aws-cdk-lib/aws-ec2";
+import { Domain, EngineVersion } from "aws-cdk-lib/aws-opensearchservice";
+import {Stack, StackProps} from "aws-cdk-lib";
+
+
+export interface opensearchServiceDomainCdkProps extends StackProps{
+  readonly version: EngineVersion,
+  readonly domainName?: string,
+  readonly dataNodeInstanceType?: string,
+  readonly dataNodes?: number,
+  readonly masterNodeInstanceType?: string,
+  readonly masterNodes?: number,
+  readonly warmInstanceType?: string,
+  readonly warmNodes?: number
+}
+
+
+export class OpensearchServiceDomainCdkStack extends Stack {
+  constructor(scope: Construct, id: string, props: opensearchServiceDomainCdkProps) {
+    super(scope, id, props);
+
+    // The code that defines your stack goes here
+
+    const domain = new Domain(this, 'Domain', {
```

Comment:
It's generally best to embed things like stage (alpha/beta/gamma/prod) and region into top-level resource names (like `'Domain'` here) as it helps w/ naming collisions when people deploy multiple into the same account.  Things like IAM Roles are global to the account and the CDK usually creates their names using a higher-level resource name as a suffix.  This means if you deploy the same stack to two regions, and the high level resources in a CDK Stack have a region-agnostic name, you can end up with collisions on global resources even though the stacks themselves are deployed to separate regions.